wgpu-core/src/device/resource.rs

```diff
@@ -149,10 +149,6 @@ impl Drop for Device {
     fn drop(&mut self) {
         resource_log!("Drop {}", self.error_ident());
 
-        if let Some(closure) = self.device_lost_closure.lock().take() {
-            closure.call(DeviceLostReason::Dropped, String::from("Device dropped."));
-        }
-
```

Comment:
nitpick: Might be good to maintain the current order of operations by explicitly dropping (viz., `let _ = self.device_lost_closure.lock().take()`). I don't know that it matters, tho.

Reply:
It shouldn't, we don't guarantee anything for the closure's `Drop` implementation. The closure's `Drop` implementation should only cleanup its own state.